spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/GenerateSymlinkFormatManifestProcedure.java

```diff
@@ -0,0 +1,142 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.spark.procedures;
+
+import java.util.Arrays;
+import org.apache.iceberg.MetadataTableType;
+import org.apache.iceberg.Partitioning;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.spark.Spark3Util;
+import org.apache.iceberg.spark.SparkTableUtil;
+import org.apache.iceberg.spark.source.SparkTable;
+import org.apache.iceberg.types.Types;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.catalyst.InternalRow;
+import org.apache.spark.sql.connector.catalog.CatalogPlugin;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import org.apache.spark.sql.connector.iceberg.catalog.ProcedureParameter;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+class GenerateSymlinkFormatManifestProcedure extends BaseProcedure {
+  private static final ProcedureParameter[] PARAMETERS = new ProcedureParameter[]{
+      ProcedureParameter.required("table", DataTypes.StringType),
+      ProcedureParameter.optional("symlink_root_location", DataTypes.StringType)
+  };
+
+  private static final StructType OUTPUT_TYPE = new StructType(new StructField[]{
+      new StructField("snapshot_id", DataTypes.LongType, false, Metadata.empty()),
+      new StructField("data_file_count", DataTypes.LongType, false, Metadata.empty())
+  });
+
+  private static final String PROCEDURE_CONTEXT = "generate symlink format manifest";
+  private static final String SYMLINK_DIRECTORY_DEFAULT = "_symlink_format_manifest";
+
+  private GenerateSymlinkFormatManifestProcedure(TableCatalog tableCatalog) {
+    super(tableCatalog);
+  }
+
+  public static SparkProcedures.ProcedureBuilder builder() {
+    return new Builder<GenerateSymlinkFormatManifestProcedure>() {
+      @Override
+      protected GenerateSymlinkFormatManifestProcedure doBuild() {
+        return new GenerateSymlinkFormatManifestProcedure(tableCatalog());
+      }
+    };
+  }
+
+  @Override
+  public ProcedureParameter[] parameters() {
+    return PARAMETERS;
+  }
+
+  @Override
+  public StructType outputType() {
+    return OUTPUT_TYPE;
+  }
+
+  @Override
+  public InternalRow[] call(InternalRow args) {
+    String tableIdent = args.getString(0);
+    Preconditions.checkArgument(tableIdent != null && !tableIdent.isEmpty(),
+        "Cannot handle an empty identifier for argument table");
```

Comment:
Nit: maybe a slightly more user-friendly error message, "table cannot be null or empty"?